amaranth/lib/data.py

```diff
@@ -789,6 +799,221 @@ def __repr__(self):
         return f"{self.__class__.__name__}({self.__layout!r}, {self.__target!r})"
 
 
+class Const(ValueCastable):
+    """A constant value viewed through the lens of a layout.
+
+    The :class:`Const` class is a constant analog of :class:`View`. It implements largely
+    the same operations, except that it deals with compile-time constants instead of arbitrary
+    :class:`Value`.
+
+    Creating a const
+    ################
+
+    A constant can be created from a dict or list of field values using :meth:`Layout.const`,
```

Comment:
I add links to the Python stdlib just to be completely unambiguous on what we mean at all times.

```suggestion
    A constant can be created from a :class:`dict` or :class:`list` of field values using :meth:`Layout.const`,
```